gorm/transaction.go

```diff
@@ -46,9 +57,31 @@ type Transaction struct {
 	mu              sync.Mutex
 	parent          *gorm.DB
 	current         *gorm.DB
+	readOnly        bool
 	afterCommitHook []func(context.Context)
 }
 
+type databaseOptions struct {
+	readOnlyReplica bool
+}
+
+type DatabaseOption func(*databaseOptions)
+
+// WithRODB returns clouser to set the readOnlyReplica flag
+func WithRODB(readOnlyReplica bool) DatabaseOption {
```

Comment:
We should add a "with function" for `sql.TxOptions` since we have existing use-cases.